src/main/java/edu/wpi/grip/core/sources/WebcamSource.java

```diff
@@ -0,0 +1,90 @@
+package edu.wpi.grip.core.sources;
+
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import edu.wpi.grip.core.Source;
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacv.Frame;
+import org.bytedeco.javacv.FrameGrabber;
+import org.bytedeco.javacv.OpenCVFrameConverter;
+import org.bytedeco.javacv.OpenCVFrameGrabber;
+
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class WebcamSource implements Source {
+    private final SocketHint<Mat> imageOutputHint = new SocketHint<Mat>("Image", Mat.class, Mat::new);
+    private final OutputSocket<Mat> outputSocket;
+    final private ScheduledThreadPoolExecutor executor;
+    private OpenCVFrameGrabber grabber;
+    private final int FRAME_RATE = 30;
```

Comment:
I'm not sure we should hard-code 30 fps here.  Are you sure there isn't a way block until a frame is ready?  Then we could just send off frames at whatever rate they come it.